app/Http/Controllers/CollectionController.php

```diff
@@ -121,6 +121,15 @@ public function viewPublicJson(Request $request, $collectionID)
                 ];
             }
         }
+
+        if(count($data['datasets']) == 0){
+            $collection->description = null; 
+            $collection->warning .= "<p>0 results found</p>";

```

Comment:
Should get rid of these mutations on the model properties. If you need to make modifications on the configurations, do it after the `$data['display']` is set.

Reply:
Updated